python-sdk/nuscenes/eval/panoptic/merge_lidarseg_and_tracking.py

```diff
@@ -0,0 +1,167 @@
+"""
+Script to generate nuScenes-panoptic predictions from nuScenes-tracking and nuScene-lidarseg predictions.
+Code written by Motional and the Robot Learning Lab, University of Freiburg.
+"""
+import argparse
+import os
+from typing import List
+
+import numpy as np
+from tqdm import tqdm
+
+from nuscenes.eval.common.loaders import load_prediction, add_center_dist
+from nuscenes.eval.common.utils import boxes_to_sensor
+from nuscenes.eval.lidarseg.utils import get_samples_in_eval_set
+from nuscenes.eval.panoptic.utils import PanopticClassMapper
+from nuscenes.eval.tracking.data_classes import TrackingBox
+from nuscenes.nuscenes import NuScenes
+from nuscenes.utils.data_classes import LidarSegPointCloud
+from nuscenes.utils.geometry_utils import points_in_box
+
+OVERLAP_THRESHOLD = 0.5
+CONFIDENCE_THRESHOLD = 0.5
+
+
+def generate_panoptic_labels(nusc: NuScenes,
+                             seg_folder: str,
+                             track_json: str,
+                             eval_set: str,
+                             out_dir: str = None,
+                             verbose: bool = False):
+    """
+    Generate NuScenes lidar panoptic predictions.
+    :param nusc: NuScenes instance.
+    :param seg_folder: Path to the directory where the lidarseg predictions are stored.
+    :param track_json: Path of the json where the tracking predictions are stored.
+    :param eval_set: Which dataset split to evaluate on, train, val or test.
+    :param out_dir: Path to save any output to.
+    :param verbose: Whether to print any output.
+    """
+    sample_tokens = get_samples_in_eval_set(nusc, eval_set)
+    num_samples = len(sample_tokens)
+    mapper = PanopticClassMapper(nusc)
+    coarse2idx = mapper.get_coarse2idx()
```

Comment:
coarse2fine ?

Reply:
This gets the mapping from coarse names to coarse indices: https://github.com/nutonomy/nuscenes-devkit/blob/cbca1b882aa4fbaf8714ea0d0897457e60a5caae/python-sdk/nuscenes/eval/lidarseg/utils.py#L188